SocialCareCaseViewerApi/V1/UseCase/FormSubmissionsUseCase.cs

```diff
@@ -56,15 +65,41 @@ public FormSubmissionsUseCase(IDatabaseGateway databaseGateway, IMongoGateway mo
             return foundSubmission?.ToDomain().ToResponse();
         }
 
-        public List<CaseSubmissionResponse> ExecuteListBySubmissionStatus(SubmissionState state)
+        public IEnumerable<CaseSubmissionResponse>? ExecuteGetByQuery(QueryCaseSubmissionsRequest request, FilterDefinition<CaseSubmission>? filter2 = null)
         {
-            var foundSubmissions = _mongoGateway
-                .LoadMultipleRecordsByProperty<CaseSubmission, SubmissionState>(_collectionName, "SubmissionState",
-                    state);
+            if (request.FormId == null && request.SubmissionStates == null)
+            {
+                throw new QueryCaseSubmissionsException("Provide at minimum one query parameter");
+            }
+
+            var builder = Builders<CaseSubmission>.Filter;
+            var filter = builder.Empty;
+
+            if (request.FormId != null)
+            {
+                filter &= Builders<CaseSubmission>.Filter.Eq(s => s.FormId, request.FormId);
```

Comment:
That's a neat way of building the filter and makes it easy to add more in the future 👍